### PR TITLE
Feature/add pull request workflow

### PR DESCRIPTION
**Pull request by tuechki**

### CODE REVIEW

File: .github/workflows/pull_request.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+name: Backend Pull Request
+
+on:
+  pull_request:
+    branches:
+      - '*'
```
Comment (Nizamski) [inline .github/workflows/pull_request.yml:6]: Maybe we should consider not to include and execute the workflow for release branches, e.g `rel-<version>`.

File: .github/workflows/pull_request.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        uses: actions/checkout@v2
+
+      - name: Set Up Java
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'adopt'
+          java-version: '17'
+
+      - name: Build with Maven
+        run: mvn clean install
```
Comment (Nizamski) [inline .github/workflows/pull_request.yml]: Wouldn't `clean package` goals be more appropriate for the case.

File: .github/workflows/pull_request.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
+name: Backend Pull Request
+
+on:
+  pull_request:
+    branches:
+      - '*'
+
+jobs:
```
Comment (Nizamski) [inline .github/workflows/pull_request.yml:18]: Consider adding `permissions` section to the workflow, in order to limit `GITHUB_TOKEN` access.

File: .github/workflows/pull_request.yml (hunk 1 above)
Comment (tuechki) [reply .github/workflows/pull_request.yml:6]: We agreeed on using tags for releases. Will mark it as resolved.

File: .github/workflows/pull_request.yml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+permissions:
+  read:
+    - actions
+    - checks
+    - contents
+    - pull-requests
+  write:
+    - checks
+  none:
```
Comment (Nizamski) [inline .github/workflows/pull_request.yml]: There is no need to describe the resources which will not have permissions. This is the default one, when there are overrides.

**Pull request merged**